src/librustc_mir/interpret/eval_context.rs

```diff
@@ -332,23 +328,29 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
     #[inline(always)]
     pub fn stack(&self) -> &[Frame<'mir, 'tcx, M::PointerTag, M::FrameExtra>] {
-        &self.stack
+        M::stack(self)
+    }
+
+    #[inline(always)]
+    pub fn stack_mut(&mut self) -> &mut Vec<Frame<'mir, 'tcx, M::PointerTag, M::FrameExtra>> {
+        M::stack_mut(self)
     }
 
     #[inline(always)]
     pub fn cur_frame(&self) -> usize {
```

Comment:
This function name has long bothered me... when seeing just `frame` and `cur_frame`, it is entirely unclear what they do. `frame` returns the current frame, after all!
So maybe `cur_frame` should be renamed to `frame_num` or `frame_idx` or so?